ui/src/App.tsx

```diff
@@ -126,7 +164,16 @@ const App = () => {
         }}
         value={endTime}
       />
-      <button onClick={handleSubmit}>Create Event</button>
+      <button
+        onClick={(e) => {
+          handleSubmit(e);
```

Comment:
This is a strange pattern you've introduced here. A cleaner way to do this would be to have a single handler that creates the entry, then fetches all the entries. The setting `newEvent` thing / useEffect you're using here is an odd way to go about it. Please refactor this so that it uses the above method I described (flattening the behavior into a single handler and getting rid of the newEvent / useEffect setup.